### ISSUE TITLE
aws_route53_record does not expand variables in name field

### ISSUE THREAD

**Issue by kkrusedp** [hashicorp/terraform#3349]
The following doesnt seem to expand variables as I would have expected.  Perhaps I am doing something wrong.
```hcl
resource "aws_route53_record" "redirect-1" {
  zone_id = "${aws_route53_zone.primary.id}"
  name = "redirect-${aws_instance.redirect.0.id}-${var.EnvName}"
  type = "A"
  ttl  = "300"
  records = [ "${aws_instance.redirect.0.private_ip}"]
}
```

renders:

```
+ aws_route53_record.redirect-1
    fqdn:      "" => "<computed>"
    name:      "" => "redirect-${aws_instance.redirect.0.id}-${var.EnvName}"
    records.#: "" => "<computed>"
    ttl:       "" => "300"
    type:      "" => "A"
    zone_id:   "" => "Z1A3R0ZG1YCDBD"
```

What I was trying to achieve with this code was setting is to generate route53 records with
redirect-instance-id-prod should produce correct records.

**Comment by radeksimko** [comment]
I believe there has been some changes in the interpolation & count which may have fixed this.

I'm not able to reproduce this anymore. I tried the following config:

```hcl
provider "aws" {
  region = "us-east-1"
}

variable "EnvName" {
	default = "test"
}

resource "aws_instance" "redirect" {
    count = 2
    ami = "ami-37bdc15d"
    instance_type = "t2.micro"
}

resource "aws_route53_zone" "primary" {
   name = "notexample.com"
}

resource "aws_route53_record" "redirect-1" {
  zone_id = "${aws_route53_zone.primary.id}"
  name = "redirect-${aws_instance.redirect.0.id}-${var.EnvName}"
  type = "A"
  ttl  = "300"
  records = [ "${aws_instance.redirect.0.private_ip}"]
}
```

and everything was created according to plan, specifically the DNS record looked like this:

```
aws_route53_record.redirect-1:
  id = Z13EN*******GQ_redirect-i-11ae0dd5-test_A
  failover =
  fqdn = redirect-i-11ae0dd5-test.notexample.com
  health_check_id =
  name = redirect-i-11ae0dd5-test
  records.# = 1
  records.988284714 = 172.31.42.136
  set_identifier =
  ttl = 300
  type = A
  weight = -1
  zone_id = Z13EN*******GQ
```

`Terraform v0.6.10-dev (b24883c52cd713164ef6dc791c124a00c859cb4a)`

Also I think that using [`element()`](https://www.terraform.io/docs/configuration/interpolation.html#element_list_index_) you can make this a bit more flexible & elegant:

```hcl
variable "count" {
	default = 2
}

resource "aws_instance" "redirect" {
    count = "${var.count}"
    ami = "ami-37bdc15d"
    instance_type = "t2.micro"
}

resource "aws_route53_record" "redirect" {
  count = "${var.count}"
  zone_id = "${aws_route53_zone.primary.id}"
  name = "redirect-${element(aws_instance.redirect.*.id, count.index)}-${var.EnvName}"
  type = "A"
  ttl  = "300"
  records = [ "${element(aws_instance.redirect.*.private_ip, count.index)}"]
}
```

Either way I'm closing this due to inability of reproduction. Feel free to reopen if you still experience this issue, ideally with a repro case (terraform code minus secrets).

**Comment by kkrusedp** [comment]
@radeksimko : I do think this may have been fixed in an update to terraform 0.6.8.   I think I reported this on 0.6.4 but it has been a while since I opened it.  I checked now and do not seem to have this issue now.
